func-mini-lib.js

```diff
@@ -0,0 +1,90 @@
+let arrLib = {};
+
+arrLib = (function (context) {
+
+    context.isArray = function (obj) {
+        return Array.isArray(obj);
+    };
+
+    context.isBoolean = function (obj) {
+        return typeof obj === 'boolean' ||
+            obj instanceof Boolean;
+    };
+
+    context.isDate = function (obj) {
+        return obj instanceof Date;
+    };
+
+    context.isNumber = function (obj) {
+        return !isNaN(parseFloat(obj)) && isFinite(obj);
+    };
+
+    context.isString = function (obj) {
+        return typeof obj === "string" ||
+            obj instanceof String;
+    };
+
+    context.isFunction = function (obj) {
+        return typeof obj === "function";
+    };
+
+    context.isUndefined = function (obj) {
+        return typeof obj === "undefined";
+    };
+
+    context.isNull = function (obj) {
+        return obj === null;
+    };
+
+    let value;
+
+    context.value = function () {
+        return value;
+    };
+
+    context.asChain = function (arr) {
```

Comment:
Chaining doesn't work as it is described in the requirements: `arrLib.asChain([1, 2]).take(1)` throws an exception

Reply:
Maybe, you forgot to call .value()